GHE_main.cpp

```diff
@@ -0,0 +1,237 @@
+//
+//  GHE_main.cpp
+//  
+//
+//  Created by Ryan Davies on 6/14/22.
+//
+
+#include <stdio.h>
+#include <string>
+#include <tuple>
+#include <vector>
+#include <numeric>
+#include <cmath>
+using namespace std;
+
+
+/* -------------------SIMPLE MEAN GHE -------------------------------
+Variables (See nomenclature section):
+H - Active borehole length (Active length of pipe)
+B - Borehole to Borehole spacing (undefined)
+D - Borehole depth below surface (undefined)
+Rb - Borehole Resistance (undefined)
+rb (undefined)
+ts - Characteristic time (way to non-dimensionalize the time) see page 12
+Ts - soil temp
+Tin - inlet temp
+mdot - mass flow rate
+cp - Specific heat (heat energy required to change temp of material)
+t_step - time step
+ks - Soil conductivity
+q_array - previous heat load data
+t_array - previous time data
+g_data - data set of g values from the JSON file
+Tout - outlet temp
+Tf - simple mean of inlet and outlet temps
+c1 - history terms
+c0 - simplication term
+qn - current heat load
+qn1 - heat load from previous run
+qi - heat at index i
+qi1 - heat at index i-1
+tn - current time step
+tn1 - previous time step
+ti - time step at index i
+ti1 - time step at index i-1
+alphas - soil thermal diffusivity (see pg 12 of Mitchell) (also see line 30 of main.py from the repo shared by Matt)
+n - current index
+i - index for summations
+total - total of summation equation. Defined in function
+g_value - output of g function. Defined in function
+ */
+
+/*TEST VALUES:
+Note: Some variables above not used because they are unique to the definition of the g function. These are marked as '(undefined)'. Becasue the g function is a provided data set they can be ignored. */
+
+// Definition of derived values and structures
+double Tout, c1, c0, Tf, qn, qi, qn1, qi1, gn, gi, ts, alphas;
```

Comment:
Drop all these variable declarations into where they are used.  Try to keep them as local as possible.  Some examples:

- If you only use a variable as a loop index, define it in the loop preamble: `for (int i = 0; i blah blah)`
- If you are creating intermediate variables to just do calculations, define them where they are used: `double c0 = 1 / (2 * M_PI * ks);`